src/components/AddTransaction/index.js

```diff
@@ -1,26 +1,50 @@
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 import {useParams} from 'react-router-dom';
 
 import useStore from '../../hooks/useStore';
 import {Form} from '../AddNewAccount/styled';
 
-export default function AddTranscation() {
+export default function AddTransaction() {
 	const [transaction, setTransaction] = useState({
 		date: '',
 		change: '',
 		note: '',
 		category: '',
 	});
 
-	const {accountID} = useParams();
+	const {accountID, transactionID} = useParams();
+	const currentBankAccount = useStore(state =>
+		state.db.find(account => account.id === accountID)
+	);
+
+	const currentTransaction = currentBankAccount.entries.find(entry => entry.id === transactionID);
+
 	const addTransaction = useStore(state => state.addTransaction);
+	const editTransaction = useStore(state => state.editTransaction);
+
+	useEffect(() => {
+		if (currentTransaction) {
+			setTransaction({
+				id: currentTransaction.id,
+				date: currentTransaction.date,
+				change: currentTransaction.change,
+				note: currentTransaction.note,
+				category: currentTransaction.category,
+			});
+		}
+	}, [currentTransaction]);
 
 	return (
 		<>
 			<Form
 				onSubmit={event => {
 					event.preventDefault();
-					addTransaction(accountID, transaction);
+					if (transactionID) {
+						editTransaction(accountID, transaction);
+						alert('edited');
+					} else {
+						addTransaction(accountID, transaction);
+					}
 					setTransaction({date: '', change: '', note: '', category: ''});
```

Comment:
Just a personal preference, however, I would write and call a function `handleSubmit` which does everything you do here. It'd just be a bit out of sight 😉

Reply:
Thanks!! We will keep that in mind once we refactor our code!